tutorials/retail_api_v2_import_products_big_query_table_java.md

```diff
@@ -0,0 +1,309 @@
+    used for requesting the Retail API.
+
+    ```bash
+    export GOOGLE_APPLICATION_CREDENTIALS=~/key.json
+    ```
+
+### Set the PROJECT_NUMBER and PROJECT_ID environment variables
+
+Because you are going to run the code samples in your own Google Cloud project, you should specify the **project_number** and **project_id** as environment variables. It will be used in every request to the Retail API.
+
+1. Find the project number and project ID in the Project Info card displayed on **Home/Dashboard**.
+
+1. Set **project_number** with the following command:
+    ```bash
+    export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+    ```
+1. Set **project_id** with the following command:
+    ```bash
+    export PROJECT_ID=<YOUR_PROJECT_ID>
+    ```
+
+### Google Cloud Retail libraries
+
+You can find the Java Google Retail library is described [here](https://googleapis.dev/java/google-cloud-retail/latest/index.html)
+
+## Clone the Retail code samples
+
+This step is required if this is the first Retail API Tutorial you run.
+Otherwise, you can skip it.
+
+Clone the Git repository with all the code samples to learn the Retail features and check them in action.
+
+<!-- TODO(ianan): change the repository link -->
+1. Run the following command in the Terminal:
+    ```bash
+    git clone https://github.com/t-karasova/grs-retail.git
+    ```
+
+    The code samples for each of the Retail services are stored in different directories.
+
+1. Go to the code samples directory - our starting point to run more commands.
+    ```bash
+    cd java-retail/samples/snippets
+    ```
+
+## Create the BigQuery table and upload products
+
+1. Before you start, build the Maven project and go to the code samples directory - our starting point to run code samples:
+   ```bash
+   cd ~/java-retail | mvn clean install -DskipTests
+   cd ~/java-retail/samples/snippets  
+   ```
+
+Before you import products to your catalog, you need to upload the data to the BigQuery table first. There are two ways to do it:
+- [Create an empty BigQuery table](https://cloud.google.com/bigquery/docs/tables#creating_an_empty_table_with_a_schema_definition) using the Retail schema and products in this table using SQL.
+- Create a BigQuery table using the prepared JSON file with products. You can define the [Retail schema](https://cloud.google.com/retail/docs/catalog#expandable-1) or use an autodetect option.
+
+To upload the data to the BigQuery table, first create a dataset, then create the table with the Retail data schema.
+Next, upload data to the table from a prepared JSON file. The data in the file should correspond the Retail schema as well.
+
+There is a <walkthrough-editor-select-regex filePath="cloudshell_open/grs-grs-retail/src/main/resources/products.json" regex="id">`resources/products.json`</walkthrough-editor-select-regex> file with valid products. It should be uploaded to the `products` dataset and `products` table.
```

Comment:
could you please replace these paths? with path to Google official repo

Reply:
done